### PR TITLE
Support for writing string values through the 'createWriteStream' method

### PR DESCRIPTION
**Pull request by headbash**
Changes directly affect the 'S3WriteStream.prototype.write' function. Now we can write strings as well as buffers. It accepts encoding types now as well.

Added functions:
-assertEncoding
-util.isString

Also added 'process.nextTick' to the 'execCb' function, so the event loop is clear before executing.

### CODE REVIEW

**Comment by DavidTPate** [comment]
Can you add some tests around this? Want to make sure that we do negative and positive tests also. Checking for invalid arguments & such.

**Comment by DavidTPate** [comment]
This fixes #25

File: lib/s3WriteStream.js [hunk 1]
```diff
@@ -186,11 +205,32 @@
...
         }
     }
 
     S3WriteStream.prototype.write = function (chunk, enc, cb) {
+        
+        var encoding;
+    
+        // parse arguments
+        if (arg1) {
+            if (util.isString(arg1)) {
```
Comment (DavidTPate) [inline lib/s3WriteStream.js]: This can just be accomplished with a typeof check. `typeof undefined` is `undefined` and `typeof null` is `object`. So the check can just be `typeof variable === 'string'`

File: lib/s3WriteStream.js [hunk 2]
```diff
@@ -186,11 +205,32 @@
...
+            process.nextTick(cb);
         }
     }
 
     S3WriteStream.prototype.write = function (chunk, enc, cb) {
+        
+        var encoding;
+    
+        // parse arguments
+        if (arg1) {
```
Comment (DavidTPate) [inline lib/s3WriteStream.js]: `arg1` and `arg2` don't exist. I see what this block is attempting to do though. Encoding is optional, so we need to see what was passed.

This block can be updated to something along the lines of:

```js
if (!cb) {
  if (typeof enc === 'function') {
    cb = enc;
    enc = undefined;
  }
}

if (enc) {
  assertEncoding(enc);
}

if (typeof chunk === 'string') {
  chunk = new Buffer(chunk, enc);
}

**Comment by headbash** [comment]
Oh gosh, I must've been so excited about my first pull request I forgot to change the arg1 and arg2 to their proper variables.

For some strange reason, looking at the source code for Node's fs module references the 'util.isString' all over the place. I kind've figured this was having some performance gain to it, as I was going to use typeof in my commit, but this thread says otherwise: https://github.com/nodejs/io.js/issues/607

I didn't find that until a couple of minutes ago.

I'm just glad you understood what I was trying to accomplish haha.

**Comment by DavidTPate** [comment]
Not a problem @headbash If you update your personal branch the updates will automatically show up in here.

**Comment by DavidTPate** [comment]
Moving this out of the `2.1.0` release milestone.

**Comment by DavidTPate** [comment]
@headbash Yeah, you started on it in this PR.
